script.js

```diff
@@ -1,97 +1,98 @@
 (function(global) {
 
+  // tempalating functing 
   var _template = function(str, param) {
     return str.replace(/\{(.*?)\}/g, function(match, token) {
       return param[token];
     });
   }
 
-  var requestData = $.getJSON('data.json').done(function(productData) {
-    var requestProductTemplate = $.get('product-template.html').done(function(productTemplateHtml) {
+  // self implemented resource loader to fulfill dependencies
+  // ** please refer to 'tiny-require.js' for annotated source code. 
+  tinyRequire(['data.json', 'product-template.html'], function(productData, productTemplateHtml) {
```

Comment:
Use the self-implemented resource loader, tinyRequire, to resolve dependencies. Inspired by `require.js`.